lib/rspec/support/object_formatter.rb

```diff
@@ -73,14 +73,22 @@ def prepare_array(array)
 
       def prepare_hash(input_hash)
         with_entering_structure(input_hash) do
-          input_hash.inject({}) do |output_hash, key_and_value|
+          sort_hash_keys(input_hash).inject({}) do |output_hash, key_and_value|
             key, value = key_and_value.map { |element| prepare_element(element) }
             output_hash[key] = value
             output_hash
           end
         end
       end
 
+      def sort_hash_keys(input_hash)
+        if input_hash.keys.all? { |k| k.is_a?(String) || k.is_a?(Symbol) }
+          Hash[input_hash.sort_by { |k, _v| k.to_s }]
+        else
+          input_hash
```

Comment:
I don't think this branch is tested?

Reply:
Oh actually it's the default so probably covered by existing

Reply:
Hi @xaviershay! Yes the raw sorting of keys is covered in object_formatter_spec.rb:38 and differ_spec.rb:248 makes sure that the diff is right. Without the sorting, there'll be incorrect additional keys in the diff which will fail differ_spec.rb:248.